src/lib.rs

```diff
@@ -476,6 +476,10 @@ pub trait HasContext {
         depth: i32,
     );
 
+    unsafe fn get_uniform_i32(&self, program: Self::Program, location: &Self::UniformLocation, v: &mut [i32]);
```

Comment:
Do you think these return `i32` and `f32` instead of accepting `&mut [i32]`/`&mut [f32]`? I guess the intention for `&mut [T]` is for this to be consistent as we expose the other `get_uniform_xyz` functions?

Reply:
tl;dr: yes. 

It's also not without prior examples in glow: `get_buffer_sub_data` and `get_tex_image_u8_slice`.

Alternatives that I can think of:

```rust
// 1
unsafe fn get_uniform_i32(&self, ...) -> Vec<i32>;

// 2
unsafe fn get_uniform_i32(&self, ...) -> i32;
unsafe fn get_uniform_i32_slice(&self, ..., v: &mut [i32]);

// 3
unsafe fn get_uniform_1_i32(&self, ...) -> i32;
unsafe fn get_uniform_2_i32(&self, ...) -> [i32; 2];
// etc
```

In the context of this function we don't know how large the uniform is which makes the first alternative pretty unappealing.

The second alternative is a small change but maybe a little quality of life. The form echoes some existing functions but I wonder how useful it will be for fetching uniforms given that it's more typical to use vectors or matrices than scalars. That quality of life makes more sense when dealing with fetching GL enums.

The third alternative is interesting. It diverges a bit from the underlying API in a way that could make porting existing code less straightforward but it could be less error-prone to use. maybe. I can see the case for that.

Reply:
Ok sounds good to me :+1: I agree that the third alternative might be interesting if the size is known, but let's proceed with the current implementation for now.